### PR TITLE
BUILD-256: add optional shared resource e2e to cso to validate shared resource related updates

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @jsafrane 

@openshift/openshift-team-build-api FYI

### CODE REVIEW

File: ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master.yaml [hunk 1]
```diff
@@ -92,6 +92,13 @@ tests:
   steps:
     cluster_profile: azure4
     workflow: openshift-e2e-azure-csi
+- as: e2e-aws-shared-resources
+  optional: true
```
Comment (coreydaley) [inline ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master.yaml:96]:
```suggestion
  optional: true
  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
```

File: ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-release-4.10.yaml [hunk 2]
```diff
@@ -93,6 +93,13 @@ tests:
   steps:
     cluster_profile: azure4
     workflow: openshift-e2e-azure-csi
+- as: e2e-aws-shared-resources
+  optional: true
```
Comment (coreydaley) [inline ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-release-4.10.yaml:97]:
```suggestion
  optional: true
  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
```

**Review by coreydaley — CHANGES_REQUESTED** [review]

**Comment by gabemontero** [comment]
e2e's look good

since I'm out tomorrow @coreydaley how about we do your skip suggestion in a separate PR and get this tagged for merge to facilitate testing in your CSO PRs.

File: ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master.yaml (hunk 1 above)
Comment (jsafrane) [reply ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master.yaml:96]: I don't think that adding it only to shared-resources is helpful. It should be either in (almost?) all CI jobs or in none.

**Comment by jsafrane** [comment]
/approve
It looks good enough to me, I let @coreydaley to judge.

File: ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master.yaml (hunk 1 above)
Comment (coreydaley) [reply ci-operator/config/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master.yaml:96]: I will update them in a subsequent pull request for all jobs, thanks!

**Pull request merged**